programs/test/benchmark.c

```diff
@@ -113,13 +113,13 @@ static void mbedtls_set_alarm(int seconds);
 #define HEADER_FORMAT   "  %-24s :  "
 #define TITLE_LEN       25
 
-#define OPTIONS                                                         \
-    "md5, ripemd160, sha1, sha256, sha512,\n"                           \
-    "sha3_224, sha3_256, sha3_384, sha3_512,\n"                         \
-    "des3, des, camellia, chacha20,\n"                                  \
-    "aes_cbc, aes_gcm, aes_ccm, aes_xts, chachapoly,\n"                 \
-    "aes_cmac, des3_cmac, poly1305\n"                                   \
-    "ctr_drbg, hmac_drbg\n"                                             \
+#define OPTIONS                                                              \
+    "md5, ripemd160, sha1, sha256, sha512,\n"                                \
+    "sha3_224, sha3_256, sha3_384, sha3_512,\n"                              \
+    "des3, des, camellia, chacha20,\n"                                       \
+    "aes_cbc, aes_cfb128, aes_cfb8, aes_gcm, aes_ccm, aes_xts, chachapoly\n" \
```

Comment:
Seems we still miss some modes such as aes-ctr, aes-ofb.
Will you add them in another PR?

Reply:
In practice, CFB, CTR and OFB aren't used much. In practice it's CBC, CCM and GCM that matter. CTR performance is somewhat useful because it's what GCM and CCM do under the hood, so it can give some information about the encryption part of those algorithms separate from the authentication part. But the GCM module and the CCM module each have their own CTR implementation, and that can significnatly affect the performance because of how they pipeline operations.

So I don't think it's particularly useful to add modes. But if we do, we should add CTR.

Reply:
This came about because of a code size improvement that had potential performance implications (#8179) so the benchmark was necessary.

Let's get this in; we can add additional modes as and when we need them.

This code was definitely useful - as a result we decided to not implement the code size reduction, as the performance impact was too great

Reply:
Agree that we can add additional modes as and when we need them.
But AES-ctr is one surprise. we know that it's part of GCM and CCM, but we also learned that it can be used independently, one example is the stream cipher DRM: both widevine and playready can use the aes-ctr as the cipher algorithm. For encryption ,seems its' performance is better than aes-cbc.